xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/resources/notification-filter-preferences.hbm.xml

```diff
@@ -48,19 +50,23 @@
 
     <property name="startingDate" type="timestamp" column="nfp_date" />
 
-    <property name="allEventTypes" type="text" column="nfp_event_types" />
-
-    <property name="user" type="string" column="nfp_user" length="768"/>
-
-    <property name="pageOnly" type="string" column="nfp_page_only" length="768" />
+    <set name="eventTypes"  table="notification_filter_prefs_eventTypes" lazy="false">
+      <key column="nfp_id" />
+      <element column="value" type="string" length="768" />
+    </set>
 
-    <property name="page" type="string" column="nfp_page" length="768" />
-
-    <property name="wiki" type="string" column="nfp_wiki" length="255" />
+    <property name="scope" column="nfp_scope" not-null="true">
```

Comment:
I think you have to define a default as otherwise hibernate will fail to add the column.

Reply:
Right, I haven't checked yet how to perform the schema migration itself: was almost wondering if it wouldn't be better to entirely rename the table to avoid problems.

Reply:
> was almost wondering if it wouldn't be better to entirely rename the table to avoid problems.

Honestly, broken for broken, it would certainly make the migration easier (just need to copy/convert rows from the old branch to the new one).

Reply:
Actually I gave it a try and it doesn't make at all the migration easier AFAICS: I need to be able to access the old table from the migration, and AFAIU I can only do that in hibernate by providing a mapping. What's even worst is that if I rename the table but keep the same class name, I break the previous migrations because the new mapping is not compatible with the old one so older migration which manipulated DefaultNotificationFilterPreferences object cannot work anymore. Now, I guess that I'll need to rewrite those migrations anyway.